src/map/CMakeLists.txt

```diff
@@ -1,45 +1,125 @@
-#
-# setup
-#
-set( MAP_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}  CACHE INTERNAL "" )
-
-
-#
-# map server
-#
-if( BUILD_SERVERS )
-message( STATUS "Creating target map-server" )
-file(GLOB MAP_HEADERS ${MAP_SOURCE_DIR}/*.hpp)
-file(GLOB MAP_SOURCES ${MAP_SOURCE_DIR}/*.cpp)
-set( DEPENDENCIES common ryml)
-set( LIBRARIES ${GLOBAL_LIBRARIES})
-set( INCLUDE_DIRS ${GLOBAL_INCLUDE_DIRS} ${COMMON_BASE_INCLUDE_DIRS} ${RA_INCLUDE_DIRS} )
-set( DEFINITIONS "${GLOBAL_DEFINITIONS} ${COMMON_BASE_DEFINITIONS}" )
-if( WITH_PCRE )
-	message( STATUS "Enabled PCRE code" )
-	set( LIBRARIES ${LIBRARIES} ${PCRE_LIBRARIES} )
-	set( INCLUDE_DIRS ${INCLUDE_DIRS} ${PCRE_INCLUDE_DIRS} )
-	set( DEFINITIONS "${DEFINITIONS} -DPCRE_SUPPORT" )
-else()
-	message( STATUS "Disabled PCRE code" )
+set(MAP_SOURCES
+	"achievement.cpp"
+	"atcommand.cpp"
+	"battle.cpp"
+	"battleground.cpp"
+	"buyingstore.cpp"
+	"cashshop.cpp"
+	"channel.cpp"
+	"chat.cpp"
+	"chrif.cpp"
+	"clan.cpp"
+	"clif.cpp"
+	"date.cpp"
+	"duel.cpp"
+	"elemental.cpp"
+	"guild.cpp"
+	"homunculus.cpp"
+	"instance.cpp"
+	"intif.cpp"
+	"itemdb.cpp"
+	"log.cpp"
+	"mail.cpp"
+	"map.cpp"
+	"mapreg.cpp"
+	"mercenary.cpp"
+	"mob.cpp"
+	"navi.cpp"
+	"npc.cpp"
+	"npc_chat.cpp"
+	"party.cpp"
+	"path.cpp"
+	"pc.cpp"
+	"pc_groups.cpp"
+	"pet.cpp"
+	"quest.cpp"
+	"script.cpp"
+	"searchstore.cpp"
+	"skill.cpp"
+	"status.cpp"
+	"storage.cpp"
+	"trade.cpp"
+	"unit.cpp"
+	"vending.cpp"
+)
```

Comment:
Consider using the following to automatically introduce all files with cpp suffixes into the MAP_SOURCES variable, so that any future addition of cpp files does not require changes to the CMakeLists.txt file.

`file(GLOB MAP_SOURCES ${MAP_SOURCE_DIR}/*.cpp)`

Reply:
I specifically do not want to do this, when we add a new file, CMake won't know to regenerate the build system.
See the [official documentation](https://cmake.org/cmake/help/latest/command/file.html?highlight=CONFIGURE_DEPENDS#filesystem)